### PR TITLE
Issue25 sqlite

### PR DESCRIPTION
**Pull request by havardf**
Run background job to fill sqlite with events, by listening to localhost NATS messages.
REST endpoint for listing all events in cache.

### CODE REVIEW

File: cmd/mms/commands.go [hunk 1]
```diff
@@ -51,7 +51,6 @@ func postEvent(hubs []mms.ProductionHub) func(*cli.Context) error {
 	return func(c *cli.Context) error {
 		productEvent := mms.ProductEvent{
 			Product:       c.String("product"),
-			ProductSlug:   c.String("product-slug"),
```
Comment (vkbo) [inline cmd/mms/commands.go:54]: Is this variable being removed? If so, it should be removed in the interface too.

**Review by vkbo — CHANGES_REQUESTED** [review]
Introduces an error in README. Changes also need to be synced with the interface. I can append those if needed.

File: README.md [hunk 2]
```diff
@@ -19,11 +19,10 @@ Shell 2:
 You should now get a printout of an MMS message.
 
 ## Build and run MMSd as docker container
-```bash
+```
 cd go-mms
 docker build -t mmsd .
-docker run -i -p 4222:4222 -p 8080:8080 -p 8088:8088 mmsd
-```
+docker run -i -p 4222:4222 -p 8080:8080 mmsd
```
[the remark this answers is not in the document]
Comment (havardf) [reply README.md:25]: I messed up a merge conflict from master I think. Will fix.

File: cmd/mms/commands.go (hunk 1 above)
Comment (mpejcoch) [reply cmd/mms/commands.go:54]: I think we should remove it, having a mandatory additional name (in addition to the file/object name) was mostly additional work for the maintainer. The "Product" should do. Perhaps we might introduce something like "Description" later.

Comment (mpejcoch) [reply cmd/mms/commands.go:54]: Where is it we need to remove it? I tried to grep for all ProductSlug occurences and remove those, maybe I have overseen some?

File: cmd/mmsd/main.go [hunk 3]
```diff
@@ -7,51 +7,73 @@ import (
...
+		log.Println("Start caching incoming events...")
 
-	webService := web.NewService(templates, staticFilesDir)
+		if err := webService.RunCache(natsURL); err != nil {
+			log.Fatalf("Caching events failed: %s", err)
+		}
+	}()
 
-	log.Println("Starting webserver for internal services ...")
+	ticker := time.NewTicker(1 * time.Hour)
```
Comment (mpejcoch) [inline cmd/mmsd/main.go:56]: Can you add a comment on what this is supposed to do?

File: internal/server/cache.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	_ "github.com/mattn/go-sqlite3" // Import sqlite3 driver for database/sql library
+	"github.com/metno/go-mms/pkg/mms"
+)
+
+const defaultDBFilePath = "/tmp/eventsCache.db"
```
Comment (mpejcoch) [inline internal/server/cache.go:16]: Find something more cross-platform

File: internal/server/cache_test.go [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+	"fmt"
+	"html/template"
+	"math/rand"
+	"os"
+	"testing"
+
+	sqlmock "github.com/DATA-DOG/go-sqlmock"
+)
+
+const staticFilesDir = "./../../static/"
```
Comment (mpejcoch) [inline internal/server/cache_test.go:14]: Can you add a FIXME, to find out a better addressing solution later.

File: internal/server/cache_test.go [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+		t.Errorf("failed to get events from mock service db: %s", err)
+	}
+
+	if len(events) != 1 {
+		t.Errorf("Expect 1 events; Got %d events", len(events))
+	}
+}
+
+func TestNewDB(t *testing.T) {
+	dbTestFile := fmt.Sprintf("/tmp/mmsdtestsqlite%d.db", rand.Int())
```
Comment (mpejcoch) [inline internal/server/cache_test.go:43]: This will probably also fail on MacOS

**Review by mpejcoch — CHANGES_REQUESTED** [review]

File: internal/server/cache.go (hunk 4 above)
Comment (havardf) [reply internal/server/cache.go:16]: This actually works on MacOS also. At least I could not reproduce it... strange.

File: internal/server/cache_test.go (hunk 6 above)
Comment (havardf) [reply internal/server/cache_test.go:43]: No, this works for me as well.

**Review by mpejcoch — APPROVED** [review]

**Review by mpejcoch — APPROVED** [review]

**Review by vkbo — APPROVED** [review]
OK!

**Pull request merged**